docs/guide/creating-variadics/README.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Creating Variadic Patch Nodes
+---
+
+# Creating Variadic Patch Nodes
+
+Declaring your patch node to be variadic allows its user to add or remove
+similar input pins dynamically. For you, it is a way to avoid creating a list
+of mechanical variations to support the various number of inputs like:
+
+* `foobar`
+* `foobar-2`
+* `foobar-3`
+* …
+* `foobar-8`
+
+Make sure you’re familiar with the [variadic nodes](../variadics/) concept
+and understand how the expansion process works.
+
+## Markers
+
+To make your patch node variadic, you should place a special marker node on it.
+Find the marker node under the `xod/patch-nodes` library. You need
+`variadic-1`, `variadic-2`, or `variadic-3`. The number defines the arity step,
+that is the group size by which the rightmost input pins replicate.
+
+The `variadic-*` node itself does nothing and is stripped out during
+compilation. It exists solely to define indefinite arity support.
+
+Consider a string join node example. We want a node which takes an arbitrary
+number (two or more) of input strings and joins them together inserting a
+delimiter between each. For example, it joins `"foo"`, `"bar"`, `"baz"` with
+delimiter `"; "` into `"foo; bar; baz"`. The delimiter is shared between all
+levels of expansion, so its input terminal goes left. We want the arity to
+increment one by one, so we use the `variadic-1` marker. Finally, a possible
+patch node implementation may look like this:
+
+![Join node implementation](./join.patch.png)
+
+## Rules
+
+To let XOD properly expand your variadic node when transpiling you must satisfy
+several natural restrictions:
+
+1. There should be no more than a single `variadic-*` marker.
+2. A variadic patch node should have at least one output.
+3. Let a patch node has 𝜄 inputs, 𝜔 outputs, and arity step 𝜎.
+   The following must be true: 𝜄 ≥ 𝜔 + 𝜎
+4. The types of inputs from the middle (indexes 𝜄 − 𝜎 − 𝜔 through 𝜄 − 𝜎 − 1)
```

Comment:
🤕 
If User already knows about "shared pin" and "variadic pin" terms, we could introduce term "middle pins" (or accumulation pins), that in the middle of shared and variadic. And input and output pairs of the "middle pins" must be of the same type even if an implicit cast is allowed in other cases.

Or something like this. :)